src/Kunstmaan/MediaBundle/Helper/MediaManager.php

```diff
@@ -131,26 +154,53 @@ public function removeMedia(Media $media)
     /**
      * @param mixed $data
      *
-     * @return Media
+     * @return mixed
      */
     public function createNew($data)
     {
-        foreach ($this->handlers as $handler) {
-            $result = $handler->createNew($data);
-            if ($result) {
-                return $result;
+        if ($this->isExtensionAllowed($data)) {
+            foreach ($this->handlers as $handler) {
+                $result = $handler->createNew($data);
+                if ($result) {
+                    return $result;
+                }
             }
         }
 
         return null;
     }
 
+    public function isExtensionAllowed(File $file): bool
+    {
+        if (!$this->limitAllowedExtesions) {
+            return true;
+        }
+
+        $mimeType = $this->mimeTypes->guessMimeType($file->getRealPath());
+        if ($mimeType === 'text/plain' && $file instanceof UploadedFile) {
+            $mimeType = $file->getClientMimeType();
+        }
+        $extensions = $this->mimeTypes->getExtensions($mimeType);
+        $extension = array_shift($extensions);
+        if (!in_array($extension, $this->allowedExtensions, true)) {
+            return false;
+        }
+
+        if (!in_array($extension, $this->imageExtensions, true)) {
```

Comment:
And I guess we can also replace the image extension config by just checking the mimetype of the upload and verifying it is image/* . As the Image constraint we are using below is just an extension of the `File` constraint with just [fixed allowed mimetypes](https://github.com/symfony/symfony/blob/367aa1df6caa2c739a58165ead81e7fe50e6599e/src/Symfony/Component/Validator/Constraints/Image.php#L60)

Reply:
Yes good idea that will simplify things.